remotespark/kernels/kernelmagics.py

```diff
@@ -18,11 +18,30 @@
 from remotespark.livyclientlib.sqlquery import SQLQuery
 from remotespark.magics.sparkmagicsbase import SparkMagicBase
 from remotespark.utils.constants import LANGS_SUPPORTED
+from remotespark.utils.sparkevents import SparkEvents
+from remotespark.utils.utils import generate_uuid, get_livy_kind
+
+
+def _event(f):
+    def wrapped(self, *args, **kwargs):
+        guid = self._generate_uuid()
+        self._spark_events.emit_magic_execution_start_event(f.__name__, get_livy_kind(self.language), guid)
+        try:
+            result = f(self, *args, **kwargs)
+        except Exception as e:
```

Comment:
The only time that happens in the magics codebase is when we're creating a session and the session creation fails -- in which case the failure would be captured in the session creation logs.